frontend/src/MuiForms/DataCardSelector.tsx

```diff
@@ -0,0 +1,155 @@
+import { Box, Chip, Stack, Typography } from '@mui/material'
+import Autocomplete from '@mui/material/Autocomplete'
+import { useTheme } from '@mui/material/styles'
+import TextField from '@mui/material/TextField'
+import { FormContextType } from '@rjsf/utils'
+import { ModelSearchResult, useListModels } from 'actions/model'
+import { debounce } from 'lodash-es'
+import { useRouter } from 'next/router'
+import { KeyboardEvent, SyntheticEvent, useCallback, useEffect, useState } from 'react'
+import { EntryKind } from 'types/types'
+
+import Loading from '../common/Loading'
+import MessageAlert from '../MessageAlert'
+
+interface DataCardSelectorProps {
+  label?: string
+  required?: boolean
+  value: string[]
+  onChange: (newValue: string[]) => void
+  formContext?: FormContextType
+  rawErrors?: string[]
+}
+
+export default function DataCardSelector(props: DataCardSelectorProps) {
+  const { onChange, value: currentValue, required, label, formContext, rawErrors } = props
+
+  const [open, setOpen] = useState(false)
+  const [dataCardListQuery, setDataCardListQuery] = useState('')
+  const [selectedDataCards, setSelectedDataCards] = useState<ModelSearchResult[]>([])
+
+  const {
+    models: dataCards,
+    isModelsLoading: isDataCardsLoading,
+    isModelsError: isDataCardsError,
+  } = useListModels(EntryKind.DATA_CARD)
+
+  const theme = useTheme()
+  const router = useRouter()
+
+  useEffect(() => {
+    if (currentValue) {
+      const updatedDataCards: ModelSearchResult[] = dataCards.filter((dataCard) => {
+        if (currentValue.includes(dataCard.id)) {
+          return dataCard
+        }
+      })
+      setSelectedDataCards(updatedDataCards)
+    }
+  }, [currentValue, dataCards])
+
+  const handleModelChange = useCallback(
+    (_event: SyntheticEvent<Element, Event>, newValues: ModelSearchResult[]) => {
+      onChange(newValues.map((value) => value.id))
+      setSelectedDataCards(newValues)
+    },
+    [onChange],
+  )
+
+  const handleInputChange = useCallback((_event: SyntheticEvent<Element, Event>, value: string) => {
+    setDataCardListQuery(value)
+  }, [])
+
+  const debounceOnInputChange = debounce((event: SyntheticEvent<Element, Event>, value: string) => {
+    handleInputChange(event, value)
+  }, 500)
+
+  if (isDataCardsError) {
+    if (isDataCardsError.status !== 413) {
```

Comment:
Why are we not displaying an error in this case?

Reply:
This was left over from the entity selector sorry! Can be removed.